src/App.tsx

```diff
@@ -1,18 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsToDisplay = items.slice(0, itemsPerPage);
+  const [visibleItems, setVisibleItems] = useState(itemsToDisplay);
```

Comment:
do not store in the state what can be calculated by existing data (itemsPerPage)
useMemo used for such approaches